fecgov/fec-cms
# Add new allocated federal/nonfederal disbursements datatable

## Summary

- Resolves #5651 
- Resolves #5652 
- Resolves #5654 


### Required reviewers

- UX
- front-end
- back-end
- content
- SMEs?

## Impacted areas of the application

- Created a feature flag (`h4_allocated_disbursements`) and wrapped the new H4 work in it
- Repo-wide: "Non-federal transfers" was changed to "**Nonfederal** transfers"
- Repo-wide: "Allocated operating expenditures - federal" changed to "Allocated operating expenditures - federal **share**"
- Repo-wide: "Allocated operating expenditures - non-federal" changed to "Allocated operating expenditures - **nonfederal** share"
- Repo-wide: "Non-federal transfers" changed to "**Nonfederal** transfers"
- Repo-wide: "Total non-federal transfers" changed to "Total **nonfederal** transfers"
- Repo-wide: "Non-federal operating expenditures" changed to "**Nonfederal** operating expenditures"
- Details panels: changed `{{city}}, {{state}}, {{zip}}` to `{{city}}, {{state}} {{zip}}` (debts, disbursements, loans, receipts)
- [Spending page](http://127.0.0.1:8000/data/browse-data/?tab=spending) - added option for "Allocated federal/nonfederal disbursements", which links to…
- The [Allocated federal/nonfederal disbursements](http://127.0.0.1:8000/data/allocated-federal-nonfederal-disbursements) page/tables and filters were created
- Text filters can now have a different type than `"text"` but it's only been implemented in this datatable's ZIP Code filter


## Screenshots

Table
<img width="1304" alt="image" src="https://github.com/fecgov/fec-cms/assets/26720877/686bdc50-7a71-4662-a7c0-77a21d07015a">

Details panel open
<img width="1304" alt="image" src="https://github.com/fecgov/fec-cms/assets/26720877/20495985-535f-4c36-ad43-cec6f6cbb900">

Committee summary page
<img width="1304" alt="image" src="https://github.com/fecgov/fec-cms/assets/26720877/60264229-b448-4384-bc84-a23bffc4baa0">

Spending page
<img width="1304" alt="image" src="https://github.com/fecgov/fec-cms/assets/26720877/876ebc35-951b-4b66-a92f-87c165b6aca6">



## Related PRs

None?

## How to test

- Check the [Known issues](#known-issues)
- pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Check the [Allocated federal/nonfederal disbursements table](http://127.0.0.1:8000/data/allocated-federal-nonfederal-disbursements/)
   - review the details panels
   - try to break the filters
- Click a committee to review its overview page
- Check the [spending](http://127.0.0.1:8000/data/browse-data/?tab=spending) page
   - "Allocated federal/nonfederal disbursements" should be listed between "Party coordinated expenditures" and  "Electioneering communications"
- Review the other items listed in [Impacted areas](#impacted-areas-of-the-application)

## Known issues:
- The date column will often be blank
- City filters need uppercase city names. Back-end is working on a fix
- There's a feature flag for this. Localhost should always show it
